### PR TITLE
fix: Personal Card Page Linting

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at 400add3</samp>

This pull request improves the handling and display of personal card transactions in the fyle-mobile-app. It updates the `PersonalCardTxn` model to support different transaction detail formats, and refactors the `personal-cards.page.ts` file to fix bugs and improve code quality.

### <samp>🤖 Generated by Copilot at 400add3</samp>

> _`PersonalCardTxn`_
> _Adapts to API changes_
> _Winter of formats_

### <samp>🤖 Generated by Copilot at 400add3</samp>

*  Import `TxnDetail` model and allow `txn_details` property to be either a string or an array of `TxnDetail` objects in `PersonalCardTxn` interface ([link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-2f7054122aee96b6a45a0f4a6077bdbb6f601de20be6a3479b8672d52a618760R1-R2), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-2f7054122aee96b6a45a0f4a6077bdbb6f601de20be6a3479b8672d52a618760L22-R24))
*  Reorder imports and add types to view child, observables, parameters, and return values in `personal-cards.page.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL1-R31), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL52-R54), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL62-R69), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL90-R92), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL104-R106), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL144-R149), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL157-R159), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL165-R167), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL176-R178), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL182-R192), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL197-R211), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL219-R225), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL227-R233), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL238-R244), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL250-R268), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL273-R279), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL285-R291), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL298-R304), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL319-R328), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL328-R337), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL348-R362), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL368-R377), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL381-R390), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL392-R401), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL399-R408), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL410-R423), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL427-R436), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL443-R462), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL473-R487), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL488-R497), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL497-R507), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL505-R514), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL585-R596), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL597-R608), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL607-R628), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL625-R644), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL633-R667), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL668-R687), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL680-R697), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL688-R705), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL717-R739), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL731-R753), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL746-R768), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL764-R786))
*  Add commas after operators to follow prettier formatting rules and improve readability ([link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL176-R178), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL182-R192), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL197-R211), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL219-R225), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL227-R233), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL238-R244), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL250-R268), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL291-R297), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL334-R343))
*  Remove unused parameters and variables to avoid unnecessary code and follow linting rules ([link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL182-R192), [link](https://github.com/fylein/fyle-mobile-app/pull/2364/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL597-R608))

## Clickup
https://app.clickup.com/t/85ztx3nye

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/personal-cards/personal-cards.page.ts [hunk 1]
```diff
@@ -494,15 +503,15 @@ export class PersonalCardsPage implements OnInit, AfterViewInit {
     }
   }
 
-  onSelectAll(event) {
-    this.selectAll = event;
+  onSelectAll(event: MatCheckboxChange | boolean): void {
```
Comment (OmkarJ13) [inline src/app/fyle/personal-cards/personal-cards.page.ts:506]: Why mixed types here? can we use `boolean` only?

File: src/app/fyle/personal-cards/personal-cards.page.ts [hunk 2]
```diff
@@ -582,7 +591,9 @@ export class PersonalCardsPage implements OnInit, AfterViewInit {
 
     await filterPopover.present();
 
-    const { data } = await filterPopover.onWillDismiss();
+    const { data } = (await filterPopover.onWillDismiss()) as {
```
Comment (OmkarJ13) [inline src/app/fyle/personal-cards/personal-cards.page.ts:594]: Can we use the existing `OverlayResponse<T>` model for this ?

File: src/app/fyle/personal-cards/personal-cards.page.ts [hunk 3]
```diff
@@ -604,11 +615,17 @@ export class PersonalCardsPage implements OnInit, AfterViewInit {
     }, 500);
   }
 
-  addNewFiltersToParams() {
+  addNewFiltersToParams(): Partial<{
+    pageNumber: number;
```
Comment (OmkarJ13) [inline src/app/fyle/personal-cards/personal-cards.page.ts:619]: Should we define a new model for this?

File: src/app/fyle/personal-cards/personal-cards.page.ts [hunk 4]
```diff
@@ -743,7 +759,13 @@ export class PersonalCardsPage implements OnInit, AfterViewInit {
 
     await selectionModal.present();
 
-    const { data } = await selectionModal.onWillDismiss();
+    const { data } = (await selectionModal.onWillDismiss()) as {
+      data: {
```
Comment (OmkarJ13) [inline src/app/fyle/personal-cards/personal-cards.page.ts:763]: We can use `OverlayResponse<T>` for this

File: src/app/core/models/personal_card_txn.model.ts [hunk 5]
```diff
@@ -19,6 +21,6 @@ export interface PersonalCardTxn {
   btxn_updated_at?: Date;
   tx_matched_at?: string;
   btxn_vendor: string;
-  txn_details?: string;
+  txn_details?: string | TxnDetail[];
```
Comment (OmkarJ13) [inline src/app/core/models/personal_card_txn.model.ts:24]: Why mixed types here? Can we use `TxnDetail[]` only?

**Review by OmkarJ13 — APPROVED** [review]
Minor comments

**Review by Chethan-Fyle — APPROVED** [review]

**Review by Dimple16 — COMMENTED** [review]
already reviewed, will skip

**Pull request merged**